src/include/daos_cont.h

```diff
@@ -717,6 +661,57 @@ int
 daos_cont_destroy_snap(daos_handle_t coh, daos_epoch_range_t epr,
 		       daos_event_t *ev);
 
+
+/**
+ * Backward compatibility code.
+ * Please don't use direclty
+ */
+
+int
+daos_cont_open2(daos_handle_t poh, const char *cont, unsigned int flags,
+		daos_handle_t *coh, daos_cont_info_t *info, daos_event_t *ev);
+
+ /**
+  * for backward compatility, support old api where a const uuid_t was used
+  * instead of a string to identify the container.
+  */
+#define daos_cont_open(poh, co, ...)					\
+	({								\
+		int _ret;						\
+		char _str[37];						\
+		const char *__str;					\
+		if (__builtin_types_compatible_p(typeof(co), char *) ||	\
+		    __builtin_types_compatible_p(typeof(co),		\
+						 const char *)) {	\
+			__str = (const char *)co;			\
+		} else {						\
+			uuid_unparse((unsigned char *)co, _str);	\
```

Comment:
I think each of the macro arguments (co and poh) when used in the macro body should be enclosed within parentheses. I guess that would apply also to the pool backward compatibility macro in daos_pool.h also. Unlikely scenario but I did something a little silly in daos_old source code, defining uuid_t myuuids[2], placing the legitimate container uuid_t value in myuuids[1], and then passing co=myuuids+1 as the argument (that is intended to specify the address &myuuids[1]). This expansion results in an incorrect address (&myuuids[0] plus one byte) passed to uuid_unparse().